scipy/interpolate/src/__fitpack.cc

```diff
@@ -290,18 +292,18 @@ fpback( /* inputs*/
     auto c = RealArray2D(cptr, nc, ydim2);
 
     // c[nc-1, ...] = y[nc-1] / R[nc-1, 0]
-    for (ssize_t l=0; l < ydim2; ++l) {
+    for (d_ssize_t l=0; l < ydim2; ++l) {
         c(nc - 1, l) = y(nc - 1, l) / R(nc-1, 0);
     }
 
     //for i in range(nc-2, -1, -1):
     //    nel = min(nz, nc-i)
     //    c[i, ...] = ( y[i] - (R[i, 1:nel, None] * c[i+1:i+nel, ...]).sum(axis=0) ) / R[i, 0]
-    for (ssize_t i=nc-2; i >= 0; --i) {
-        ssize_t nel = std::min(nz, nc - i);
-        for (ssize_t l=0; l < ydim2; ++l){
+    for (d_ssize_t i=nc-2; i >= 0; --i) {
```

Comment:
It's very minor thing but if you define all variables at the top Fortran style, it becomes much easier for the compiler to detect unused vars and also reduces the mistyped vars acting stupid with implicit casts.